tests/core/utilities/test_event_interface.py

```diff
@@ -18,7 +19,7 @@
 
 def test_access_event_with_no_abi(w3):
     contract = w3.eth.contract()
-    with pytest.raises(NoABIEventsFound):
+    with pytest.raises(NoABIFound):
```

Comment:
We should also check that `ABIFunctionNotFound` is still raised where it used to be raised if we want to go that route.

Reply:
Events now behave similar to functions, so the logic for the exceptions match what functions have. That exception is raised for functions when a contract is initialized without an `abi`, then any attempt to use functions would raise the `NoABIFound` exception.

I agree we should not have breaking changes for this. Maybe one day this could be addressed as a breaking change. I can create a new issue to track that.

For this, I will change the logic for events so that `NoABIEventsFound` is raised and it wont be breaking.

Reply:
Going to mark this resolved. We cleared this up that a contract with no ABI at all should indeed raise the `NoABIFound` exception. This test case seems to be expecting the wrong exception here so this is technically a bugfix.